File: pixles-core-rust/src/models/asset.rs
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub struct Asset {
    /// Asset ID
    pub id: Uuid,
    /// Album ID
    pub album_id: Option<String>,
    /// Owner ID
    pub owner_id: String,
    /// File extension (e.g., "png", "mp4", "json")
    /// Do NOT prepend with a dot (`.`)
    /// String is case-sensitive
    pub ext: String,
}

impl Asset {
    pub fn new(album_id: Option<String>, owner_id: String, ext: String) -> Self {
        Self {
            id: Uuid::now_v7(),
            album_id,
            owner_id,
            ext,
        }
    }
}

File: pixles-core/src/commonMain/kotlin/com/justin13888/pixles/models/Asset.kt
package com.justin13888.pixles.models

import kotlin.uuid.Uuid

/**
 * Represents a media asset in Pixles.
 *
 * @param id Asset ID
 * @param albumId Album ID this asset belongs to, if any
 * @param ownerId Owner ID
 * @param ext File extension (e.g., "png", "mp4", "json"). Do NOT prepend with a dot.
 *            String is case-sensitive.
 */
data class Asset(
    val id: Uuid,
    val albumId: String?,
    val ownerId: String,
    val ext: String,
) {
    companion object {
        fun new(albumId: String?, ownerId: String, ext: String): Asset =
            Asset(id = Uuid.random(), albumId = albumId, ownerId = ownerId, ext = ext)
    }
}
